### PR TITLE
Add ErrorCode property to RequestFailedException

### LINKED ISSUE #8143: Core: Add RequestFailedException.ErrorCode
https://apiview.azurewebsites.net/Assemblies/Review/b48340c11dab4e88b3b2bd269bc03661#Azure.Storage.StorageRequestFailedException

Decide if we're going to add an ErrorCode to the base RequestFailedException so we can optionally remove the StorageRequestFailedException.

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/8143

### CODE REVIEW

File: sdk/core/Microsoft.Extensions.Azure/samples/Startup.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public void ConfigureServices(IServiceCollection services)
                 builder.ConfigureDefaults((options, provider) =>  options.AddPolicy(provider.GetService<DependencyInjectionEnabledPolicy>(), HttpPipelinePosition.PerCall));
 
                 builder.AddBlobServiceClient(Configuration.GetSection("Storage"))
-                        .WithVersion(BlobClientOptions.ServiceVersion.V2018_11_09);
+                        .WithVersion(BlobClientOptions.ServiceVersion.V2019_02_02);
```
Comment (pakrym) [inline sdk/core/Microsoft.Extensions.Azure/samples/Startup.cs]: This fixes the build.

File: sdk/storage/Azure.Storage.Common/src/StorageRequestFailedException.cs [hunk 2]
```diff
@@ -73,15 +68,8 @@ public StorageRequestFailedException(Response response, string message, Exceptio
             : base(
                   response?.Status ?? throw Errors.ArgumentNull(nameof(response)),
                   CreateMessage(response, message ?? response?.ReasonPhrase, errorCode, additionalInfo),
-                  innerException)
+                  GetErrorCode(response, errorCode), innerException)
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/src/StorageRequestFailedException.cs]: Nit - should we also pass this into `CreateMessage` two lines up?

**Review by tg-msft — APPROVED** [review]
Awesome - we'll remove the Storage type with #8144

File: sdk/storage/Azure.Storage.Common/src/StorageRequestFailedException.cs (hunk 2 above)
Comment (pakrym) [reply sdk/storage/Azure.Storage.Common/src/StorageRequestFailedException.cs]: You didn't do it for that parameter so I also didn't :)

**Pull request merged**